### PR TITLE
Address minor warnings

### PR DESCRIPTION
**Pull request by OleksiiBondar**
Minor clean up

### CODE REVIEW

**Comment by shakuzen** [comment]
I was originally confused reviewing this because I did not have any of these warnings. Then I realized that the Eclipse compiler generates these warnings but javac (used by IntelliJ by default) does not. I'm torn between improving the developer experience for those using Eclipse and not wanting to add things to code that the official build's (`./gradlew build`) compiler doesn't warn.

**Comment by OleksiiBondar** [comment]
@shakuzen 
Should we apply the same logic for other suppressWarning annotation types as well? I.e. 
```
@SuppressWarnings("ConstantConditions")
@SuppressWarnings("EqualsWhichDoesntCheckParameterClass")
@SuppressWarnings("FieldCanBeLocal")
```
Having these does not make any difference for me in Eclipse and in console output while compiling.
As an example, I've removed several mentioned above `@SuppressWarning` annotations from `micrometer-registry-atlas` module and ran build command: 
`./gradlew build -p implementations/micrometer-registry-atlas/ --debug`
I'm not seeing any difference, any new warnings. Am I missing something?

File: implementations/micrometer-registry-statsd/src/test/java/io/micrometer/statsd/StatsdMeterRegistryTest.java [hunk 1]
```diff
@@ -289,13 +289,15 @@ void customNamingConvention() {
     @Test
     void counterIncrementDoesNotCauseStackOverflow() {
         StatsdMeterRegistry registry = new StatsdMeterRegistry(configWithFlavor(StatsdFlavor.ETSY), clock);
-        new LogbackMetrics().bindTo(registry);
+        try (LogbackMetrics logbackMetrics = new LogbackMetrics()) {
```
Comment (jkschneider) [inline implementations/micrometer-registry-statsd/src/test/java/io/micrometer/statsd/StatsdMeterRegistryTest.java]: I think this may actually be breaking.

**Comment by shakuzen** [comment]
Sorry for the inaction on this. It may be easier to start from scratch again. Rebasing is a bit painful now. If you're still interested in contributing the changes, would you please rebase or open a new pull request?

On a case-by-case basis we can add `@SuppressWarnings` to improve the developer experience in IDEs. I think all the maintainers are using IntelliJ, so it will be harder for us to manually maintain suppress warnings specific to Eclipse. But I don't think it's harmful other than clutter to have them for contributors using Eclipse.

**Comment by shakuzen** [comment]
Closing for now. If you rebase the pull request to get rid of the merge conflicts, we can reopen and pick this back up. Thanks.

**Pull request closed without merging**